packages/flutter/lib/src/widgets/router.dart

```diff
@@ -663,6 +691,7 @@ class _RouterState<T> extends State<Router<T>> {
 
   Future<void> _rebuild([void value]) {
     setState(() {/* routerDelegate is ready to rebuild */});
+    _maybeNeedToReportRouteInformation();
```

Comment:
Can you add a test for this? Create a RouterInformation Parser that redirects and verify the new routeinformation is reported to the engine.

Reply:
Added.